sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryServicesImpl.java

```diff
@@ -1002,6 +1002,24 @@ public List<TableDataInsertAllResponse.InsertErrors> call() throws Exception {
               throw e;
             }
             String errorReason = errorInfo.getReason();
+
+            // Return errors so that we can handle even NotFound and AccessDenied with the retry
+            // policy
+            if (ApiErrorExtractor.INSTANCE.itemNotFound(e)
+                || ApiErrorExtractor.INSTANCE.accessDenied(e)) {
+              LOG.warn("Ignore the error: " + e.getMessage());
+              List<TableDataInsertAllResponse.InsertErrors> errors = new ArrayList<>();
+              TableDataInsertAllResponse.InsertErrors errorDetail =
+                  new TableDataInsertAllResponse.InsertErrors();
+              ErrorProto errorProto = new ErrorProto();
+              errorProto.setReason(errorReason);
+              errorProto.setMessage("Item not found or access denied: " + e.getMessage());
+              errorDetail.setIndex(0L);
+              errorDetail.setErrors(Collections.singletonList(errorProto));
+              errors.add(errorDetail);
+              return errors;
+            }
```

Comment:
If it encounters errors as NotFound and AccessDenied when inserting data into BigQuery, it doesn't return the response format to get `insertErrors`. In the BigQuery SDK in Java, it raises `IOException` by errors as `NotFound` and `AccessDenied`. The implementation would be a workaround to capture the exception. I'm not sure it is going to a right direction. But, it is an easy way to change the behavior so that we can retrieve errors with the retry policy.

https://cloud.google.com/bigquery/docs/reference/rest/v2/tabledata/insertAll